app/src/main/java/org/asteroidos/sync/services/SynchronizationService.java

```diff
@@ -31,198 +33,241 @@
-        silentModeService = new SilentModeService(getApplicationContext());
+        mPrefs = getSharedPreferences(MainActivity.PREFS_NAME, Context.MODE_PRIVATE);
+        String defaultDevMacAddr = mPrefs.getString(MainActivity.PREFS_DEFAULT_MAC_ADDR, "");
+        if (defaultDevMacAddr.equals("")) return;
+        String defaultLocalName = mPrefs.getString(MainActivity.PREFS_DEFAULT_LOC_NAME, "");
+        BluetoothDevice device = BluetoothAdapter.getDefaultAdapter().getRemoteDevice(defaultDevMacAddr);
+        device.createBond();
+        mBleMngr.connect(device)
+                .useAutoConnect(true)
+                .timeout(100000)
+                .retry(3, 200)
+                .done(device1 -> Log.d(TAG, "Connected to " + device1.getName()))
+                .fail((device2, error) -> Log.e(TAG, "Failed to connect to " + device.getName() +
+                        " with error code: " + error))
+                .enqueue();
+    }
+
+    final void handleDisconnect() {
+        if (mBleMngr == null) return;
+        if (mState == STATUS_DISCONNECTED) return;
 
-        mDevice.connect();
+        bleServices.forEach(IService::unsync);
+        mBleMngr.abort();
+        mBleMngr.disconnect().enqueue();
     }
 
-    void handleDisconnect() {
-        if(mDevice == null) return;
-        if(mState == STATUS_DISCONNECTED) return;
-        mScreenshotService.unsync();
-        mWeatherService.unsync();
-        mNotificationService.unsync();
-        mMediaService.unsync();
-        mTimeService.unsync();
-        mDevice.disconnect();
-        silentModeService.onDisconnect();
+    final void handleSetDevice(BluetoothDevice device) {
+        SharedPreferences.Editor editor = mPrefs.edit();
+        Log.d(TAG, "handleSetDevice: " + device.toString());
+        editor.putString(MainActivity.PREFS_DEFAULT_MAC_ADDR, device.getAddress());
+        mDevice = device;
+        String name = mDevice.getName();
+        try {
+            Message answer = Message.obtain(null, MSG_SET_LOCAL_NAME);
+            answer.obj = name;
+            replyTo.send(answer);
+            replyTo.send(Message.obtain(null, MSG_SET_STATUS, mState, 0));
+        } catch (RemoteException | NullPointerException ignored) {
+        }
+        editor.putString(MainActivity.PREFS_DEFAULT_LOC_NAME, name);
+        editor.apply();
     }
 
-    void handleReqBattery() {
-        if(mDevice == null) return;
-        if(mState == STATUS_DISCONNECTED) return;
-        mDevice.read(AsteroidUUIDS.BATTERY_UUID, new BleDevice.ReadWriteListener()
-        {
-            @Override public void onEvent(ReadWriteEvent result)
-            {
-                if(result.wasSuccess())
-                    try {
-                        replyTo.send(Message.obtain(null, MSG_SET_BATTERY_PERCENTAGE, result.data()[0], 0));
-                    } catch (RemoteException | NullPointerException ignored) {}
+    final void handleUpdate() {
```

Comment:
Maybe this could be renamed into something more descriptive ? handleUpdateConnectionStatus ?